### ISSUE TITLE
Slot in server component causes hydration issues

### ISSUE THREAD

**Issue by manniL** [nuxt/nuxt#25011]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.9.0
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, experimental
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

[StackBlitz](https://stackblitz.com/edit/nuxt-issues-25011)

0. Enable `experimental.componentIslands` by setting it to `'local'`
1. Create a `CtaButton.server.vue` component in `components`
2. Render a `slot` in there:

```vue
<template>
  <button>
    <slot />
  </button>
</template>
```

3. Render it in `app.vue`

```vue
<template>
  <div>
    <CtaButton>Test</CtaButton>
  </div>
</template>
```

4. See hydration error in the console

### Describe the bug

A hydration error occurs.

**Comment by huang-julien** [comment]
Seems like there's an issue with the replace server side. 
I'm currently working on a giant refactor for slots and the `NuxtIslandResponse` which will probably fix this

**Comment by huang-julien** [comment]
@manniL it should be fixed in 3.9.2. Just tested on your reproduction, i don't have the hydration issue anymore
